src/NumericalAlgorithms/Interpolation/SpanInterpolator.cpp

```diff
@@ -13,18 +13,21 @@
 namespace intrp {
 
 std::complex<double> SpanInterpolator::interpolate(
-    const gsl::span<double>& source_points,
-    const gsl::span<std::complex<double>>& values,
+    const gsl::span<const double>& source_points,
+    const gsl::span<const std::complex<double>>& values,
     const double target_point) const noexcept {
-  ComplexDataVector view{values.data(), values.size()};
+  // the operation below to get the real and imag parts does not alter the
+  // contents of the span, so the const-cast is safe.
+  const ComplexDataVector view{const_cast<std::complex<double>*>(values.data()),
+                               values.size()};
   DataVector real_part = real(view);
```

Comment:
these are const; also there are two allocation per target point; please at least file an issue to investigate the cost of this.